tokens/usecase/tokens_usecase.go

```diff
@@ -357,6 +363,64 @@ func (t *tokensUseCase) getChainScalingFactorMut(precision int) (osmomath.Dec, b
 	return result, ok
 }
 
+var (
+	jobIsRunning   bool
+	jobIsrunningMu sync.Mutex
+)
+
+// StartAsRoutineIfNotRunning starts the job in a new goroutine if it is not already running.
+func StartAsRoutineIfNotRunning(job func()) bool {
+	jobIsrunningMu.Lock()
+	if jobIsRunning {
+		jobIsrunningMu.Unlock()
+		return false
+	}
+	jobIsRunning = true
+	jobIsrunningMu.Unlock()
+
+	go func() {
+		jobIsrunningMu.Lock()
+		job()
+		jobIsRunning = false
+		jobIsrunningMu.Unlock()
+	}()
+
+	return true
+}
+
+// ExecuteOnTickInterval executes a callback at specified intervals.
+func ExecuteOnTickInterval(tick uint64, interval int, callback func()) {
+	if tick%uint64(interval) == 0 {
+		callback()
+	}
+}
+
+// UpdateAssetsAtHeightInterval updates assets at specified height intervals by running fn.
+// Function fn will be launched as separate goroutine.
```

Comment:
TODO: add docs; if `fn` ( `LoadTokensFromChainRegistryFunc` ) is already executing it will not be called twice.